hail/python/hailtop/batch/batch.py

```diff
@@ -90,7 +90,8 @@ def __init__(self,
                  default_memory: Optional[str] = None,
                  default_cpu: Optional[str] = None,
                  default_storage: Optional[str] = None,
-                 default_timeout: Optional[Union[float, int]] = None):
+                 default_timeout: Optional[Union[float, int]] = None,
+                 default_shell: str = '/bin/bash'):
```

Comment:
if we specify default_shell here we can avoid specifying /bin/bash for every backend (although maybe we want different shell defaults?), and save a job._shell None check per backend, but as you said at the disadvantage of a small amount of visibility into user behavior (although I don't see where we log Batch() invocations)

Reply:
Sleeping on this, the point is that if you know there is user input, you can add another default (e.g an environment variable).  Also, None matches the rest of the code, which is always a benefit.